launch/client.py

```diff
@@ -247,28 +296,178 @@ def _upload_data(self, data: bytes) -> str:
             requests.put(s3_path, data=data)
         return raw_bundle_url
 
-    def create_model_bundle_from_dirs(
+    def _get_bundle_url_from_base_paths(self, base_paths: List[str]) -> str:
+        tmpdir = tempfile.mkdtemp()
+        try:
+            zip_path = os.path.join(tmpdir, "bundle.zip")
+            _zip_directories(zip_path, base_paths)
+            with open(zip_path, "rb") as zip_f:
+                data = zip_f.read()
+        finally:
+            shutil.rmtree(tmpdir)
+
+        raw_bundle_url = self._upload_data(data)
+        return raw_bundle_url
+
+    def _upload_model_bundle(
+        self,
+        load_model_fn: Callable,
+        load_predict_fn: Callable,
+        bundle_metadata: Dict[str, Any],
+    ):
+        bundle = dict(load_model_fn=load_model_fn, load_predict_fn=load_predict_fn)
+        bundle_metadata["load_predict_fn"] = inspect.getsource(load_predict_fn)  # type: ignore
+        bundle_metadata["load_model_fn"] = inspect.getsource(load_model_fn)  # type: ignore
+        serialized_bundle = cloudpickle.dumps(bundle)
+        bundle_location = self._upload_data(data=serialized_bundle)
+        return bundle_location
+
+    def _upload_schemas(self, request_schema: Type[BaseModel], response_schema: Type[BaseModel]) -> str:
+        model_definitions = get_model_definitions(
+            request_schema=request_schema,
+            response_schema=response_schema,
+        )
+        model_definitions_encoded = json.dumps(model_definitions).encode()
+        return self._upload_data(model_definitions_encoded)
+
+    def create_model_bundle_from_callable_v2(
+        self,
+        *,
+        model_bundle_name: str,
+        load_predict_fn: Callable[[LaunchModel_T], Callable[[Any], Any]],
```

Comment:
@Anirudh-Scale You may want to take a look at this PR. I think it's still orthogonal to the Q2-ish cloudpickle API changes you're thinking of, but note that this `create_model_bundle_from_callable_v2` is being created. What this means for the rest of the @scaleapi/launch-eng crew - just be aware that we may be deprecating `load_predict_fn` in favor of something else at some point in the future.